### PR TITLE
[Enhancement](DOE) Step1: Fe generates the DSL and is used to explain

### PR DESCRIPTION
**Pull request by stalary**
# Proposed changes

Issue Number: https://github.com/apache/incubator-doris/issues/9835

## Problem Summary:
For the first step, I will only change FE and then change BE once I make sure the DSL is ok.

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/external/elasticsearch/EsUtil.java [hunk 1]
```diff
@@ -87,12 +105,134 @@ public static JSONObject getJsonObject(JSONObject jsonObject, String key, int fr
...
-            throw new DdlException(String.format("fail to parse %s, %s = %s, `%s` should be like 'true' or 'false'， value should be double quotation marks", name, name, property, name));
+            throw new DdlException(String.format("fail to parse %s, %s = %s, `%s` should be like 'true' or 'false'， "
+                    + "value should be double quotation marks", name, name, property, name));
+        }
+    }
+
+    /**
+     * Doris expr to es dsl.
+     **/
+    public static QueryBuilder convertToEsDsl(Expr expr) {
```
[the remark this answers is not in the document]
Comment (stalary) [reply fe/fe-core/src/main/java/org/apache/doris/external/elasticsearch/EsUtil.java:124]: done.

**Comment by stalary** [comment]
convertToEsDsl move the generate dsl of `exec/es/es_query_builder.cpp` up.

File: fe/fe-core/src/main/java/org/apache/doris/external/elasticsearch/EsUtil.java [hunk 2]
```diff
@@ -87,12 +106,156 @@ public static JSONObject getJsonObject(JSONObject jsonObject, String key, int fr
...
+                    return QueryBuilders.boolQuery().mustNot(QueryBuilders.termQuery(column, value));
+                case GE:
+                    return QueryBuilders.rangeQuery(column).gte(value);
+                case GT:
+                    return QueryBuilders.rangeQuery(column).gt(value);
+                case LE:
+                    return QueryBuilders.rangeQuery(column).lte(value);
+                case LT:
+                    return QueryBuilders.rangeQuery(column).lt(value);
+                case EQ_FOR_NULL:
```
Comment (Userwhite) [inline fe/fe-core/src/main/java/org/apache/doris/external/elasticsearch/EsUtil.java:177]: move it up, like this
``` 
      case EQ:
      case EQ_FOR_NULL:
```

File: fe/fe-core/src/main/java/org/apache/doris/external/elasticsearch/EsUtil.java [hunk 3]
```diff
@@ -87,12 +106,156 @@ public static JSONObject getJsonObject(JSONObject jsonObject, String key, int fr
...
+            return decimalLiteral.getValue();
+        } else if (expr instanceof FloatLiteral) {
+            FloatLiteral floatLiteral = (FloatLiteral) expr;
+            return floatLiteral.getValue();
+        } else if (expr instanceof IntLiteral) {
+            IntLiteral intLiteral = (IntLiteral) expr;
+            return intLiteral.getValue();
+        } else if (expr instanceof StringLiteral) {
+            StringLiteral stringLiteral = (StringLiteral) expr;
+            return stringLiteral.getStringValue();
```
Comment (Userwhite) [inline fe/fe-core/src/main/java/org/apache/doris/external/elasticsearch/EsUtil.java:257]: We also need largeint literal

**Comment by Userwhite** [comment]
maybe we can send dsl to be， and add check to compare :)

**Comment by stalary** [comment]
> maybe we can send dsl to be， and add check to compare :)

That's a good idea. I'll do it later.

File: fe/fe-core/src/main/java/org/apache/doris/external/elasticsearch/QueryBuilders.java [hunk 4]
```diff
@@ -172,19 +183,65 @@ abstract static class QueryBuilder {
...
+         **/
+        public String toJson() {
+            StringWriter writer = new StringWriter();
+            try {
+                JsonGenerator gen = mapper.getFactory().createGenerator(writer);
+                this.toJson(gen);
+                gen.flush();
+                gen.close();
+            } catch (IOException e) {
+                LOG.warn("QueryBuilder toJson error");
```
Comment (qidaye) [inline fe/fe-core/src/main/java/org/apache/doris/external/elasticsearch/QueryBuilders.java:198]: The error should print to warn log.

Comment (stalary) [reply fe/fe-core/src/main/java/org/apache/doris/external/elasticsearch/QueryBuilders.java:198]: Step one, in order to be compatible, fe generate dsl can not affect exist logic, what do you think?

File: fe/fe-core/src/main/java/org/apache/doris/external/elasticsearch/EsUtil.java [hunk 5]
```diff
@@ -210,4 +231,216 @@ private static void resolveDocValuesFields(SearchContext searchContext, JSONObje
...
+        String filterPath = docValueMode ? "filter_path=_scroll_id,hits.total,hits.hits._score,hits.hits.fields"
+                : "filter_path=_scroll_id,hits.hits._source,hits.total,hits.hits._id";
+        if (limit <= 0) {
+            StringBuilder initScrollUrl = new StringBuilder();
+            StringBuilder nextScrollUrl = new StringBuilder();
+            initScrollUrl.append("/").append(index);
+            if (StringUtils.isNotBlank(type)) {
+                initScrollUrl.append("/").append(type);
+            }
+            initScrollUrl.append("/_search?scroll=5m&").append(filterPath).append("&terminate_after=")
```
Comment (qidaye) [inline fe/fe-core/src/main/java/org/apache/doris/external/elasticsearch/EsUtil.java:432]: Should the `5m` be configurable?

**Review by qidaye — APPROVED** [review]
LGTM

**Pull request merged**